### PR TITLE
Fix runKernel with Grid Reductions

### PR DESCRIPTION
**Pull request by kevinstephano**

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/kernel.cpp [hunk 1]
```diff
@@ -339,7 +339,10 @@ size_t numberOfReductionSegments(
       (z_block ? 1 : grid_dim.z);
 }
 
-std::array<size_t, 2> gridReductionTempBufferSizes(CudaKernel* entry) {
+std::array<size_t, 2> gridReductionTempBufferSizes(
+    CudaKernel* entry,
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/kernel.cpp:343]: Can we drop `entry` as it doesn't seem to be used anymore?

Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/kernel.cpp:343]: We still need it for traversing graph few lines down.

Comment (naoyam) [reply torch/csrc/jit/codegen/cuda/kernel.cpp:343]: Ah, my bad. I'd suggest to pass just `fusion` or `fusion->exprs(true)`.

**Review by jjsjann123 — APPROVED** [review]
LGTM.

**Pull request merged**